metabase/metabase
# Replace old DatePicker in dashboard & search parameters

https://github.com/metabase/metabase/issues/50477

Replaces the old DatePicker in parameters with a modern version from the query builder. There should be no UX changes apart from more "modern" components. 

How to verify dashboards:
- Add `Date` -> `All Options` parameter and connect it to a card
- Try to set all possible date filters
- Note that `Is empty` and `Not empty` operators are disabled for dashboards

<img width="657" alt="Screenshot 2024-12-23 at 10 18 34" src="https://github.com/user-attachments/assets/70cb98a7-be31-433b-9078-c42f26b88545" />
<img width="678" alt="Screenshot 2024-12-23 at 10 18 42" src="https://github.com/user-attachments/assets/90ad1477-8a8b-427d-815a-9be04cd360ce" />


How to verify search:
- Search -> View results
- Use date filters on the right
- All date picker options should work
- Note that `Exclude` date filters are disabled for search

<img width="1728" alt="Screenshot 2024-12-20 at 18 25 45" src="https://github.com/user-attachments/assets/3d88408f-5e51-48a0-8e24-89c4087a4c2c" />
<img width="362" alt="Screenshot 2024-12-20 at 18 27 05" src="https://github.com/user-attachments/assets/93f785ca-2ac8-4a15-b86f-ff23feeff227" />
